builtin/logical/transit/backend.go

```diff
@@ -100,13 +113,45 @@ func GetCacheSizeFromStorage(ctx context.Context, s logical.Storage) (int, error
 	return size, nil
 }
 
-func (b *backend) invalidate(_ context.Context, key string) {
+// Update cache size and get policy
+func (b *backend) GetPolicy(ctx context.Context, polReq keysutil.PolicyRequest, rand io.Reader) (retP *keysutil.Policy, retUpserted bool, retErr error) {
+	if b.lm.GetUseCache() && b.cacheSizeChanged {
```

Comment:
Since cacheSizeChanged can be written in a different thread, you should protect the reading of it with the configMutex as well.  Take a read lock where it's read, and a write lock where it's written.  Take care to unlock the read half before you take the write lock to change it.